### DISCUSSION TITLE
Bulk import local images

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#120] [Ideas]
I would like to bulk import local images from a specific folder (in subdirectory of the wiki using relative paths) into a single tiddler.  Is it possible?
Images shall be imported in [img[]] or [[]] format which shall be configurable.

 I would like to make album for some life events. This is intended for that project.

**Comment by arunnbabu81** [comment]
https://kookma.github.io/TW-Scripts/#Create%20External%20Images%20from%20A%20Directory

https://groups.google.com/g/tiddlywiki/c/1yTi65sc8D8/m/0ru-PhMJBAAJ

Found this in Google group. Is something similar possible?

**Reply by buggyj** [reply]
You would need to clip each link separately - a bit like with pdfs

**Reply by arunnbabu81** [reply]
> You would need to clip each link separately - a bit like with pdfs

That will be too much time consuming when there are many images in a folder.

**Reply by buggyj** [reply]
you could clip the html and extract the urls with a regex

**Reply by arunnbabu81** [reply]
> you could clip the html and extract the urls with a regex

How to do it? Can you give an example?

**Reply by buggyj** [reply]
Try this - I have only tested on linux.
[allimages.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/11183344/allimages.json.zip)

**Reply by arunnbabu81** [reply]
> Try this - I have only tested on linux. [allimages.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/11183344/allimages.json.zip)

This works. But I need only the relative path of the image files and need images as links without img styling [[]].
So can seperate rules be made to use either absolute or relative paths of filenames and [[]] or [img[]] links for images

**Reply by buggyj** [reply]
you should have relative paths if the images are in a subdir under where the tiddlywiki is.

You can set the `%img` to `""` to remove the `img`

![image](https://user-images.githubusercontent.com/2855795/230738042-aba1ea92-f59c-4718-a415-f874a5f6dfc1.png)

**Reply by arunnbabu81** [reply]
> you should have relative paths if the images are in a subdir under where the tiddlywiki is.
> 
> You can set the `%img` to `""` to remove the `img`
> 
> ![image](https://user-images.githubusercontent.com/2855795/230738042-aba1ea92-f59c-4718-a415-f874a5f6dfc1.png)

![image](https://user-images.githubusercontent.com/67494083/230738516-c07ea0bb-67f5-45e2-8b6b-b587124caa85.png)


![image](https://user-images.githubusercontent.com/67494083/230738533-3b0a8c88-ddab-4a68-b4f0-9b0a3e5e70a1.png)


I am not getting the relative path even though the images are in a subdirectory of the TW folder as shown in the screenshots

**Reply by buggyj** [reply]
It looks like there is a difference in the  case of the drive letter `C:`  vs `c:`

**Reply by arunnbabu81** [reply]
> It looks like there is a difference in the case of the drive letter `C:` vs `c:`

![image](https://user-images.githubusercontent.com/67494083/230739613-da86f731-562c-49bc-86c6-622ea3ae75f7.png)

![image](https://user-images.githubusercontent.com/67494083/230739628-3902969b-645d-4a22-8c6d-eac487e9e71c.png)

I corrected the case of the drive letter. Still there is some extra prefix `file:./` in the image file names. How to remove it

**Reply by buggyj** [reply]
try this version

[allimages2.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/11184162/allimages2.json.zip)

**Comment by arunnbabu81** [comment]
If I use this rule
![image](https://user-images.githubusercontent.com/67494083/230741826-1d1ee7be-af10-41b0-849d-03ae73948d06.png)
extra prefix `file:./` in the image file name is seen

![image](https://user-images.githubusercontent.com/67494083/230741848-1a8d337c-3805-42a1-a099-9065594b334c.png)

But if i use this rule

![image](https://user-images.githubusercontent.com/67494083/230741883-43a598bc-fdaa-4f20-8015-3641378a767c.png)


I get correct relative file path

![image](https://user-images.githubusercontent.com/67494083/230741898-a9c5b7de-ef51-4636-b17a-89eaf29bc8bb.png)

**Reply by buggyj** [reply]
The uri scheme (`file:` in this case) is required to form an external link (with double square brackets)- see https://tiddlywiki.com/#Linking%20in%20WikiText.

**Reply by arunnbabu81** [reply]
> The uri scheme (`file:` in this case) is required to form an external link (with double square brackets)- see https://tiddlywiki.com/#Linking%20in%20WikiText.

But why it works in the other example?

**Comment by arunnbabu81** [comment]
Also the images are not displayed when wrapped in Macy widget as shown [here](https://github.com/buggyj/tiddlyclip-plugin/discussions/118#discussioncomment-5518919) if the link with relative path has `file`prefix . This occurs when the I try to open those tiddlers in other devices.

**Reply by buggyj** [reply]
?? what other devices

**Reply by arunnbabu81** [reply]
> ?? what other devices 
> 
> 

I am away from home for a few more hours. I tested in windows laptop and Android phone.....in windows laptop the `file:` prefix works but in Android phone it doesn't. I have to test my macbook when I reach home

**Reply by buggyj** [reply]
I have not tried to use locally stored media on andoid.

Here is a new vesion to allow the removal of `file` url prefix
[allimages3.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/11185112/allimages3.json.zip)

**Reply by arunnbabu81** [reply]
> I have not tried to use locally stored media on andoid.
> 
> Here is a new vesion to allow the removal of `file` url prefix [allimages3.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/11185112/allimages3.json.zip)

Now its working. Thank you @buggyj

**Comment by arunnbabu81** [comment]
@buggyj I want to wrap the `<<macy-spotlight """ """>>` around the image links imported by this rule. How to modify the rule to do it ?

**Reply by buggyj** [reply]
add

`$text=<<macy-spotlight """((*$text*))""">>`

at the end of the `bj/tc/doextractlinks` tiddler
